### PR TITLE
epik: add User-Agent

### LINKED ISSUE #2268: epik: returns 404
### Welcome

- [X] Yes, I'm using a binary release within 2 latest releases.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### What did you expect to see?

I expected to renew/run a cert with the dns backend of epik.

### What did you see instead?

404 error

### How do you use lego?

Binary

### Reproduction steps

this is my script, 
```
#!/bin/bash

export EPIK_SIGNATURE=[REDACTED]
export EPIK_PROPAGATION_TIMEOUT=900
export EPIK_POLLING_INTERVAL=30
export EPIK_TTL=300
/usr/local/bin/lego --email [REDACTED] --dns epik --domains *.[REDACTED] renew
/usr/local/bin/lego --email [REDACTED] --dns epik --domains *.[REDACTED] --domains [REDACTED] renew
```

### Version of lego

```console
lego version 4.18.0 linux/amd64
```

### Logs

<details>

```console

2024/09/08 [REDACTED] [INFO] [[REDACTED]] acme: Trying renewal with -2699 hours remaining
2024/09/08 [REDACTED] [INFO] [[REDACTED], *.[REDACTED]] acme: Obtaining bundled SAN certificate
2024/09/08 [REDACTED] [INFO] [*.[REDACTED]] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/401146463086
2024/09/08 [REDACTED] [INFO] [[REDACTED]] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/401146463096
2024/09/08 [REDACTED] [INFO] [*.[REDACTED]] acme: use dns-01 solver
2024/09/08 [REDACTED] [INFO] [[REDACTED]] acme: Could not find solver for: tls-alpn-01
2024/09/08 [REDACTED] [INFO] [[REDACTED]] acme: Could not find solver for: http-01
2024/09/08 [REDACTED] [INFO] [[REDACTED]] acme: use dns-01 solver
2024/09/08 [REDACTED] [INFO] [*.[REDACTED]] acme: Preparing to solve DNS-01
2024/09/08 [REDACTED] [INFO] [[REDACTED]] acme: Preparing to solve DNS-01
2024/09/08 [REDACTED] [INFO] [*.[REDACTED]] acme: Cleaning DNS-01 challenge
2024/09/08 [REDACTED] [WARN] [*.[REDACTED]] acme: cleaning up failed: epik: 404: <!doctype html>
<html lang="en-US" class="no-js">
<head>
    <meta charset="UTF-8">
    <title> Page not found - Epik</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32"
          href="https://www.epik.com/wp-content/themes/epik-redesign/src/img/favicon.png">
    <meta name="theme-color" content="#ffffff">
    <meta name='robots' content='noindex, follow'/>
</head>
<body class="error404">
<div id="body-content" class="wrapper">
    <main class="e404-page" role="main" id="main-content">
        <div class="container-fluid">
            <div class="row">
                <div class="col-lg-6 pt-5 pb-5 text-side">
                    <div class="content-wrapper">
                        <div class="header-wrapper">
                            <h1>404 <span class="e404-header-smaller">Page not found</span></h1>
                        </div>
                        <div class="text-wrapper">
                            <p>Whoops, the link you clicked may be broken or the page may have been removed.</p>
                            <p>Return to the <a href="https://www.epik.com">homepage</a> to get back on track or <a
                                    href="/contact/">contact us</a> for assistance.</p>
                        </div>
                    </div>
                </div>
                <div class="d-none d-lg-block col-lg-6 background-side"></div>
            </div>
        </div>
    </main>
</div>
<script>(function(){function c(){var b=a.contentDocument||a.contentWindow.document;if(b){var d=b.createElement('script');d.innerHTML="window.__CF$cv$params={r:'8c02a2027b63d7a7',t:'MTcyNTgzNzI0NS4wMDAwMDA='};var a=document.createElement('script');a.nonce='';a.src='/cdn-cgi/challenge-platform/scripts/jsd/main.js';document.getElementsByTagName('head')[0].appendChild(a);";b.getElementsByTagName('head')[0].appendChild(d)}}if(document.body){var a=document.createElement('iframe');a.height=1;a.width=1;a.style.position='absolute';a.style.top=0;a.style.left=0;a.style.border='none';a.style.visibility='hidden';document.body.appendChild(a);if('loading'!==document.readyState)c();else if(window.addEventListener)document.addEventListener('DOMContentLoaded',c);else{var e=document.onreadystatechange||function(){};document.onreadystatechange=function(b){e(b);'loading'!==document.readyState&&(document.onreadystatechange=e,c())}}}})();</script></body>
</html>
 
2024/09/08 [REDACTED] [INFO] [[REDACTED]] acme: Cleaning DNS-01 challenge
2024/09/08 [REDACTED] [WARN] [[REDACTED]] acme: cleaning up failed: epik: 404: <!doctype html>
<html lang="en-US" class="no-js">
<head>
    <meta charset="UTF-8">
    <title> Page not found - Epik</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32"
          href="https://www.epik.com/wp-content/themes/epik-redesign/src/img/favicon.png">
    <meta name="theme-color" content="#ffffff">
    <meta name='robots' content='noindex, follow'/>
</head>
<body class="error404">
<div id="body-content" class="wrapper">
    <main class="e404-page" role="main" id="main-content">
        <div class="container-fluid">
            <div class="row">
                <div class="col-lg-6 pt-5 pb-5 text-side">
                    <div class="content-wrapper">
                        <div class="header-wrapper">
                            <h1>404 <span class="e404-header-smaller">Page not found</span></h1>
                        </div>
                        <div class="text-wrapper">
                            <p>Whoops, the link you clicked may be broken or the page may have been removed.</p>
                            <p>Return to the <a href="https://www.epik.com">homepage</a> to get back on track or <a
                                    href="/contact/">contact us</a> for assistance.</p>
                        </div>
                    </div>
                </div>
                <div class="d-none d-lg-block col-lg-6 background-side"></div>
            </div>
        </div>
    </main>
</div>
<script>(function(){function c(){var b=a.contentDocument||a.contentWindow.document;if(b){var d=b.createElement('script');d.innerHTML="window.__CF$cv$params={r:'8c02a202dcdbd7a7',t:'MTcyNTgzNzI0NS4wMDAwMDA='};var a=document.createElement('script');a.nonce='';a.src='/cdn-cgi/challenge-platform/scripts/jsd/main.js';document.getElementsByTagName('head')[0].appendChild(a);";b.getElementsByTagName('head')[0].appendChild(d)}}if(document.body){var a=document.createElement('iframe');a.height=1;a.width=1;a.style.position='absolute';a.style.top=0;a.style.left=0;a.style.border='none';a.style.visibility='hidden';document.body.appendChild(a);if('loading'!==document.readyState)c();else if(window.addEventListener)document.addEventListener('DOMContentLoaded',c);else{var e=document.onreadystatechange||function(){};document.onreadystatechange=function(b){e(b);'loading'!==document.readyState&&(document.onreadystatechange=e,c())}}}})();</script></body>
</html>
 
2024/09/08 23:14:05 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/401146463086
2024/09/08 23:14:06 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/401146463096
2024/09/08 23:14:06 error: one or more domains had a problem:
[*.[REDACTED]] [*.[REDACTED]] acme: error presenting token: epik: 404: <!doctype html>
<html lang="en-US" class="no-js">
<head>
    <meta charset="UTF-8">
    <title> Page not found - Epik</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32"
          href="https://www.epik.com/wp-content/themes/epik-redesign/src/img/favicon.png">
    <meta name="theme-color" content="#ffffff">
    <meta name='robots' content='noindex, follow'/>
</head>
<body class="error404">
<div id="body-content" class="wrapper">
    <main class="e404-page" role="main" id="main-content">
        <div class="container-fluid">
            <div class="row">
                <div class="col-lg-6 pt-5 pb-5 text-side">
                    <div class="content-wrapper">
                        <div class="header-wrapper">
                            <h1>404 <span class="e404-header-smaller">Page not found</span></h1>
                        </div>
                        <div class="text-wrapper">
                            <p>Whoops, the link you clicked may be broken or the page may have been removed.</p>
                            <p>Return to the <a href="https://www.epik.com">homepage</a> to get back on track or <a
                                    href="/contact/">contact us</a> for assistance.</p>
                        </div>
                    </div>
                </div>
                <div class="d-none d-lg-block col-lg-6 background-side"></div>
            </div>
        </div>
    </main>
</div>
<script>(function(){function c(){var b=a.contentDocument||a.contentWindow.document;if(b){var d=b.createElement('script');d.innerHTML="window.__CF$cv$params={r:'8c02a1eb09b4d7a7',t:'MTcyNTgzNzI0Mi4wMDAwMDA='};var a=document.createElement('script');a.nonce='';a.src='/cdn-cgi/challenge-platform/scripts/jsd/main.js';document.getElementsByTagName('head')[0].appendChild(a);";b.getElementsByTagName('head')[0].appendChild(d)}}if(document.body){var a=document.createElement('iframe');a.height=1;a.width=1;a.style.position='absolute';a.style.top=0;a.style.left=0;a.style.border='none';a.style.visibility='hidden';document.body.appendChild(a);if('loading'!==document.readyState)c();else if(window.addEventListener)document.addEventListener('DOMContentLoaded',c);else{var e=document.onreadystatechange||function(){};document.onreadystatechange=function(b){e(b);'loading'!==document.readyState&&(document.onreadystatechange=e,c())}}}})();</script></body>
</html>

[[REDACTED]] [[REDACTED]] acme: error presenting token: epik: 404: <!doctype html>
<html lang="en-US" class="no-js">
<head>
    <meta charset="UTF-8">
    <title> Page not found - Epik</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="icon" type="image/png" sizes="32x32"
          href="https://www.epik.com/wp-content/themes/epik-redesign/src/img/favicon.png">
    <meta name="theme-color" content="#ffffff">
    <meta name='robots' content='noindex, follow'/>
</head>
<body class="error404">
<div id="body-content" class="wrapper">
    <main class="e404-page" role="main" id="main-content">
        <div class="container-fluid">
            <div class="row">
                <div class="col-lg-6 pt-5 pb-5 text-side">
                    <div class="content-wrapper">
                        <div class="header-wrapper">
                            <h1>404 <span class="e404-header-smaller">Page not found</span></h1>
                        </div>
                        <div class="text-wrapper">
                            <p>Whoops, the link you clicked may be broken or the page may have been removed.</p>
                            <p>Return to the <a href="https://www.epik.com">homepage</a> to get back on track or <a
                                    href="/contact/">contact us</a> for assistance.</p>
                        </div>
                    </div>
                </div>
                <div class="d-none d-lg-block col-lg-6 background-side"></div>
            </div>
        </div>
    </main>
</div>
<script>(function(){function c(){var b=a.contentDocument||a.contentWindow.document;if(b){var d=b.createElement('script');d.innerHTML="window.__CF$cv$params={r:'8c02a2008c63d7a7',t:'MTcyNTgzNzI0NS4wMDAwMDA='};var a=document.createElement('script');a.nonce='';a.src='/cdn-cgi/challenge-platform/scripts/jsd/main.js';document.getElementsByTagName('head')[0].appendChild(a);";b.getElementsByTagName('head')[0].appendChild(d)}}if(document.body){var a=document.createElement('iframe');a.height=1;a.width=1;a.style.position='absolute';a.style.top=0;a.style.left=0;a.style.border='none';a.style.visibility='hidden';document.body.appendChild(a);if('loading'!==document.readyState)c();else if(window.addEventListener)document.addEventListener('DOMContentLoaded',c);else{var e=document.onreadystatechange||function(){};document.onreadystatechange=function(b){e(b);'loading'!==document.readyState&&(document.onreadystatechange=e,c())}}}})();</script></body>
</html>


```

</details>

### Go environment (if applicable)

<details>

```console
$ go version && go env
# paste output here
```

</details>

### PR DESCRIPTION
**Pull request by ldez**
Fixes #2268

### CODE REVIEW

File: providers/dns/epik/internal/client.go [hunk 1]
```diff
@@ -15,6 +15,8 @@ import (
 
 const defaultBaseURL = "https://usersapiv2.epik.com/v2"
 
+const defaultUserAgent = "goacme/lego"
```
Comment (dmke) [inline providers/dns/epik/internal/client.go:18]: In #2291, this is `go-acme/lego`, and `ourUserAgent` in `acme/api/internal/sender/useragent.go` is `xenolf-acme/<version>`. I haven't checked, but likely other providers might also set user agents.

Can we just export and use `sender.OurUserAgent`, or does this introduce a circular dependency?

**Review by dmke — CHANGES_REQUESTED** [review]

File: providers/dns/epik/internal/client.go (hunk 1 above)
Comment (ldez) [reply providers/dns/epik/internal/client.go:18]: This is a special case: https://github.com/go-acme/lego/issues/2268#issuecomment-2394007004

Comment (ldez) [reply providers/dns/epik/internal/client.go:18]: About the user `xenolf-acme/<version>`: this is related to Let's Encrypt to ease the detection from their side and for historical reasons it's difficult to change it.

I already think of having a shared user-agent but `sender.OurUserAgent` is difficult to share.
Maybe I should create something inside the `dns01` package.

Comment (dmke) [reply providers/dns/epik/internal/client.go:18]: `xenolf-acme/<version>` doesn't contain `go-`, so my point still stands :)

Comment (ldez) [reply providers/dns/epik/internal/client.go:18]: This kind of special case is why it's difficult to really share pieces between implementations, there are some unexpected details (ex: `Accept` header, etc.) that break API (yes there is at least one implementation that doesn't support `Accept`... DNS API is a world by itself :smile: ).

Comment (ldez) [reply providers/dns/epik/internal/client.go:18]: I don't really want to spread `xenolf-acme/<version>` because it's just an old thing.

Comment (ldez) [reply providers/dns/epik/internal/client.go:18]: I will add a comment to explain why it's not the same user agent as the other.

**Review by dmke — APPROVED** [review]

File: providers/dns/epik/internal/client.go (hunk 1 above)
Comment (dmke) [reply providers/dns/epik/internal/client.go:20]: LGTM

**Pull request merged**